pkg/kubelet/prober/worker.go

```diff
@@ -167,6 +176,7 @@ func (w *worker) doProbe() (keepGoing bool) {
 
 	if w.containerID.String() != c.ContainerID {
 		if !w.containerID.IsEmpty() {
+			metrics.ProberResults.Delete(proberResultsMetricLabels)
```

Comment:
This is evoked whenever a container is restarted, but you're not tracking container identity (i.e. no containerID label, nor do I think there should be). This should go in the worker cleanup routine instead: https://github.com/kubernetes/kubernetes/pull/58827/files#diff-1f8bc3c65e0eaaad18129c3a9616603aR112